src/module/vue/components/asset/asset-ability.vue

```diff
@@ -0,0 +1,187 @@
+<template>
+	<IronCheckbox
+		:icon-checked="iconChecked"
+		:icon-unchecked="iconUnchecked"
+		:checked="ability.enabled"
+		:class="{ [$style.wrapper]: true, [$style.hoverFx]: toggle && canUpdate }"
+		:readonly="!(toggle && canUpdate)"
+		:icon-switch-class="$style.bullet"
+		:content-hover-fx="false"
+		@change="toggleAbility">
+		<WithRolllisteners
+			element="div"
+			:class="$style.rulesText"
+			class="flexcol"
+			@moveclick="moveclick"
+			v-html="$enrichHtml(ability.description)" />
+		<Clock
+			v-if="ability.hasClock"
+			:class="$style.clock"
+			:wedges="ability.clockMax"
+			:ticked="ability.clockTicks"
+			:readonly="readonlyClock ?? !canUpdate"
+			@click="updateClock($event)" />
+	</IronCheckbox>
+</template>
+
+<script lang="ts" setup>
+import { computed } from 'vue'
+import { IronswornSettings } from '../../../helpers/settings'
+import type { AssetAbility } from '../../../item/itemtypes'
+import Clock from '../clock.vue'
+import type { IconSwitchState } from '../icon/icon-common'
+import { FontAwesome } from '../icon/icon-common'
+import IronCheckbox from '../input/iron-checkbox.vue'
+import WithRolllisteners from '../with-rolllisteners.vue'
+
+const props = defineProps<{
+	/**
+	 * Can the asset ability be toggled?
+	 */
+	toggle?: boolean
+	/**
+	 * Render the asset ability clock as read-only?
+	 */
+	readonlyClock?: boolean
+	/** NYI */
+	readonlyFields?: boolean
+	ability: AssetAbility
+	/**
+	 * A function that updates the asset ability via merge. This can safely be omitted if no interactivity is required.
+	 */
+	updateFn?: (delta: Partial<AssetAbility>) => Promise<unknown>
+}>()
+
+// if there's no provided update function, assume it's a statically rendered ability; the clock and the checkbox can't be manipulated
+const canUpdate = computed(() => !!props.updateFn)
+
+const iconChecked = computed<IconSwitchState>(() => {
+	switch (IronswornSettings.starforgedToolsEnabled) {
+		case true:
+			return {
+				icon: 'fa:hexagon',
+				props: {
+					rotate: FontAwesome.Rotate['90deg'],
+					family: FontAwesome.Family.Solid
+				}
+			}
+		default:
+			return {
+				icon: 'fa:circle',
+				props: { family: FontAwesome.Family.Solid }
+			}
+	}
```

Comment:
This is a weird way to do this:

```suggestion
	if (IronswornSettings.starforgedToolsEnabled) {
		return {
			icon: 'fa:hexagon',
			props: {
				rotate: FontAwesome.Rotate['90deg'],
				family: FontAwesome.Family.Solid
			}
		}
	}
	return {
		icon: 'fa:circle',
		props: { family: FontAwesome.Family.Solid }
	}
```

Reply:
yeah, fair. i think my reasoning (can't be sure, it was pretty late) was that it'd be easier to add additional options to a switch, but that'd probably be handled in a totally different theme file anyways.